src/foam/u2/view/RichChoiceView.js

```diff
@@ -617,7 +617,15 @@ foam.CLASS({
     {
       name: 'onDataUpdate',
       code: function() {
-        if ( this.data === undefined ) this.clearSelection();
+        if ( this.data === undefined ) {
+           this.clearSelection();
+           return;
+        }
+        this.sections.forEach(section => {
+          section.dao.find(this.data).then(result => {
+            this.fullObject_ = result;
```

Comment:
Wouldn't `fullObject_` be just overridden to the result from the last find?

Reply:
Well there should be only one find since richChoiceView doesn't have multi-select support afaik

Reply:
Can we make sure that `fullObject_` will be set to the result from valid find though when you have multiple sections as find is done async?

Reply:
We can check that `result != undefined` before setting `fullObject_ == result` but other than that, no way to make sure I guess